mlflow/store/sqlalchemy_store.py

```diff
@@ -206,43 +221,33 @@ def log_param(self, run_uuid, param):
             self._get_or_create(SqlParam, run_uuid=run_uuid, key=param.key,
                                 value=param.value)
         except sqlalchemy.exc.IntegrityError:
-            raise MlflowException('changing parameter {} value is not allowed'.format((run_uuid,
-                                                                                      param)),
+            self.session.rollback()
+            old_param, _ = self._get_or_create(SqlParam, run_uuid=run_uuid, key=param.key)
```

Comment:
Could we use the already-gotten param from `run`? It's possible we get an IntegrityError for some reason that does not imply that _get_or_create will actually get (i.e., it's possible in some future state of the database that it retries the create, and fails because value is null).

Reply:
aah! rite ... we could totally use param from run's data. Why dint i think of that? Would obviate the need for this ^rollback^ above as well..